src/Sema.zig

```diff
@@ -15758,6 +15758,15 @@ fn zirAsm(
             .ComptimeInt => arg.* = try sema.coerce(block, Type.usize, uncasted_arg, src),
             .ComptimeFloat => arg.* = try sema.coerce(block, Type.f64, uncasted_arg, src),
             else => {
+                if (!uncasted_arg_ty.hasRuntimeBits(mod)) {
+                    const input_op_src: LazySrcLoc = .{
+                        .asm_input_op = .{
+                            .asm_node_offset = extra.data.src_node,
+                            .input_index = @intCast(u32, arg_i),
```

Comment:
I don't like this cast but I see we do this in a bunch of places. Is this something we just accept in the compiler, undefined behavior when given absurdly large values?

Reply:
Here are the rules I follow:
- if something would break earlier in practice (e.g. here, having >2^32 args requires either a tuple too big to really represent or a too-large source file), it's fine to use `@intCast`
- otherwise, if you're in Sema casting to a usize (u64 -> usize is a very common case), use Sema.usizeCast
- otherwise, just @intCast if you need, we can look at it down the line if we want

Reply:
I kinda wanted to do `math.cast(u32, arg_i) orelse sema.fail("too many operands", .{})` but I don't really wanna create such a big file to test it either. It's not satisfactory but let's just leave it as-is for now then...